zilliqa/src/consensus.rs

```diff
@@ -1028,6 +1037,27 @@ impl Consensus {
         Ok(Some((proposal, broadcasted_transactions)))
     }
 
+    pub fn propose_pending_block(&mut self) -> Result<Block> {
+        let (proposal, applied_transactions) = self
```

Comment:
I can't see this function used anywhere? What does it do and why is it necessary?

Reply:
Does `preview_content` not let you view the mempool without `&mut self`?